### DISCUSSION TITLE
Error in Korean title of video after switching to premium version

### DISCUSSION THREAD

**Discussion by bbiggiee** [aclap-dev/video-downloadhelper#655] [Help and Q&A]
### Your question

An error occurs in the combination of characters in the Korean title of the every downloaded video.

for example) 안녕하세요.mp4 (O) -> ㅇㅏㄴㄴㅕㅇㅎㅏㅅㅔㅇㅛ.mp4 (X)

Please solve this 

### Addon & CoApp version and details

version: 9.0.1.22
target: google
channel: stable
lang: ko
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"aa69b6be91764d5e"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "used_popup_button": true,
    "download_directory": "C:\\Users\\dlxor\\Downloads",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive"
        }
    },
    "concurrent_downloads_max": 10,
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": false,
    "open_count_store": 6,
    "successfull_dl": 11
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I don't speak Korean so I'm not sure I'd notice errors in the names of files.  But I can see in your example above that the Korean characters are changing to things that are not characters in any language I know of.  My suspicion is this has something to do with a global language setting on your system, perhaps something to do with code pages.  If you give an example URL of a download that did this to you, I could compare your results with mine.  I will say that I have recently downloaded something that was a string of Japanese characters & that worked fine.  So I think there is hope.  But let me see an example & we'll take it from there.

**Reply by bbiggiee** [reply]
Heres korean video link -> https://m.blog.naver.com/joro7155/clip/4470661

If you download the video and check the title, you can see that the title has been changed to an 네이버 블로그 -> ㄴㅔㅇㅣㅂㅓㅂㅡㄹㄹㅗㄱㅡ and saved.
Korean characters are basically combined with consonants and vowels to complete a single letter. 
for example) ㅎ(h)+ㅏ(o)+ㅅ(t) = 핫(hot)
However, currently, programs do not combine consonants and vowels.

This phenomenon does not only occur in HLS format, it corresponds to the Korean language of all files downloaded.
I don't know if this is due to a premium upgrade or if an update happened in the meantime.

**Reply by GrampaWildWilly** [reply]
I went to your URL & VDH told me this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/1d56845e-8b8e-40a4-83f7-008491291985)

So I started the download & it told me this:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/2478dbba-1b2e-4164-b66e-157ad81b826d)

It looks like I'm seeing the same problem you are.

I did a Google search & quickly found this page:

https://learn.microsoft.com/en-us/windows/win32/intl/code-page-identifiers

On that page, it told me code page 949 is for ANSI/OEM Korean (Unified Hangul Code).  There are other code pages for Korean listed on that page but you would have to tell me if any of them is preferable to this one.  I used this information to execute this at a command prompt:

```
C:\>cd "Mozilla Firefox"

C:\Mozilla Firefox>chcp 949
Active code page: 949

C:\Mozilla Firefox>firefox.exe
```

It didn't make any difference.  I think we're going to need help from someone who knows more than I do.  That would be @paulrouget Paul.  He's the developer.

This one is a simple MP4.  You say HLS acts differently.  It might help with problem diagnosis to offer an example that uses HLS.

**Reply by bbiggiee** [reply]
Thank you for your interest. 
Unfortunately, I don't know much about the code, so I'm afraid I can't help you. 
I really hope this issue will be resolved soon.

https://vod.kbs.co.kr/index.html?source=episode&sname=vod&stype=vod&program_code=T2019-0063&program_id=PS-2024081841-01-000&broadcast_complete_yn=N&local_station_code=00&section_code=04&section_sub_code=02
This is a Korean state broadcasting site that is streamed on HLS. But, i'm not sure if this is a better code page...

Also i tried all other formats of video files, not just HLS, but they all had the same problem.
This is a sudden problem. Maybe I can only pray for it to be fixed in the next update.

**Reply by GrampaWildWilly** [reply]
I'm geoblocked on that site.  I don't think I can be of any help here.

**Reply by bbiggiee** [reply]
Finally i find how to fix!
simply change setting the use LEGACY UI XD
This is an older UI, so there are minor inconveniences, but it's not a big problem^^
thanks for your help!

**Comment by paulrouget** [comment]
Hello. That's something I'm working on at the moment, will fix this soon I hope!

**Comment by bbiggiee** [comment]
I fixed the problem! 
When I change to use the old version LEGACY UI in settings, the title is output normally!

**Comment by c29979** [comment]
I experience the same phenomenon with Japanese titles. The version is 9.0.2.3 stable in Google Chrome on Win10. It's not a problem with a specific video or a specific website. In all the websites, dakutens (dots of e.g. ば) and han-dakutens (circle of e.g. ぱ) of kana characters and all the other punctuations (such as ・「」) are dropped.

For example, the name of the saved file becomes `THE NEW GATE 第10話 瘴魔テーモン.mp4` where I expect `THE NEW GATE 第10話 「瘴魔（デーモン）」.mp4`. Please notice that `デ` is somehow converted to `テ`. Also, `「」` and `()` are dropped, and as a result, words are connected in a wrong way.

It did work as expected before. I don't download videos every day, but as far as I can tell, the last successful one was on 2024-05-21 and the first failure was on 2024-05-28. The problem must have started to happen with an update between those two dates.

The first error reported in #673 must be the same issue.

**Reply by c29979** [reply]
#551 seems to be the same issue as well.

**Reply by GrampaWildWilly** [reply]
Also #838.

**Reply by c29979** [reply]
The problem hasn't yet been fixed. #671 and #738 are also the same issue.
